server/src/proxy/hotspot.rs

```diff
@@ -0,0 +1,432 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! hotspot recorder
+use std::{fmt::Write, sync::Arc, time::Duration};
+
+use ceresdbproto::storage::{
+    PrometheusQueryRequest, RequestContext, SqlQueryRequest, WriteRequest,
+};
+use common_util::{runtime::Runtime, timed_task::TimedTask};
+use log::{info, warn};
+use serde::{Deserialize, Serialize};
+use spin::Mutex;
+pub use spin::Mutex as SpinMutex;
+use tokio::sync::mpsc::{self, Sender};
+
+use crate::proxy::{hotspot_lru::HotspotLru, util};
+
+type QueryKey = String;
+type WriteKey = String;
+const TAG: &str = "hotspot autodump";
+const RECODER_CHANNEL_CAP: usize = 64 * 1024;
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(default)]
+pub struct Config {
+    /// Max items size for query hotspot
+    query_cap: Option<usize>,
+    /// Max items size for write hotspot
+    write_cap: Option<usize>,
+    dump_interval: Duration,
```

Comment:
Maybe we should call it as `auto_dump_interval` to be consistent with the two options below.

Reply:
ok